php-checks/src/main/java/org/sonar/php/checks/GenericExceptionCheck.java

```diff
@@ -48,87 +48,90 @@
+      context().newIssue(KEY, MESSAGE).tree(namespaceName);
+    }
+
+    super.visitThrowStatement(tree);
   }
 
   @Override
-  public void visitNode(AstNode astNode) {
-    if (astNode.is(PHPGrammar.NAMESPACE_STATEMENT)) {
-      parseNamespace(astNode);
-      return;
+  public void visitUseClause(UseClauseTree tree) {
+    String name = tree.namespaceName().name().text();
+
+    if (tree.namespaceName().namespaces().isEmpty() && RAW_EXCEPTIONS.contains(name)) {
+      importedGenericExceptions.add(tree.alias() != null ? tree.alias().text() : name);
     }
-    if (astNode.is(PHPGrammar.USE_DECLARATION)) {
-      parseUse(astNode);
-      return;
+
+    super.visitUseClause(tree);
+  }
+
+  @Override
+  public void visitNamespaceStatement(NamespaceStatementTree tree) {
+    if (tree.namespaceName() != null) {
+      currentNamespace = tree.namespaceName().fullName();
     }
-    if (isBaseException(astNode)) {
-      getContext().createLineViolation(this, MESSAGE, astNode);
+
+    super.visitNamespaceStatement(tree);
+
+    // Delimited namespace with curly braces
+    if (tree.openCurlyBrace() != null) {
+      currentNamespace = "";
     }
   }
 
-  private void parseNamespace(AstNode astNode) {
-    AstNode namespaceNode = astNode.getFirstChild(PHPGrammar.NAMESPACE_NAME);
-    namespace = getStringValue(namespaceNode);
+  private boolean isGenericException(NamespaceNameTree namespaceName) {
+    return isImportedGenericException(namespaceName) || isGlobalNamespaceGenericException(namespaceName);
+
   }
 
-  private void parseUse(AstNode astNode) {
-    AstNode namespaceNode = astNode.getFirstChild(PHPGrammar.NAMESPACE_NAME);
-    String namespaceValue = getStringValue(namespaceNode);
-    for (String exc : RAW_EXCEPTIONS) {
-      if (namespaceValue.equals(exc)) {
-        AstNode idNode = astNode.getFirstChild(PHPGrammar.IDENTIFIER);
-        String alias = idNode != null ? getStringValue(idNode) : namespaceValue;
-        uses.add(alias);
-      }
-    }
+  private boolean isImportedGenericException(NamespaceNameTree namespaceName) {
+    return importedGenericExceptions.contains(namespaceName.fullName());
   }
 
-  private boolean isBaseException(AstNode throwNode) {
-    AstNode classNameNode = throwNode.getFirstDescendant(PHPGrammar.CLASS_NAME);
-    if (classNameNode == null) {
-      return false;
-    }
-    String className = getStringValue(classNameNode);
-    if (className.startsWith(SEPARATOR)) {
-      return RAW_EXCEPTIONS.contains(className.substring(1));
-    }
-    if (namespace.isEmpty()) {
-      return RAW_EXCEPTIONS.contains(className);
-    }
+  private boolean isGlobalNamespaceGenericException(NamespaceNameTree namespaceName) {
+    return isFromGlobalNamespace(namespaceName) && RAW_EXCEPTIONS.contains(namespaceName.name().text());
+  }
 
-    return uses.contains(className);
+  private boolean isFromGlobalNamespace(NamespaceNameTree namespaceName) {
+      return !namespaceName.hasQualifiers()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 5. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)